### DISCUSSION TITLE
Apache as Reverse-Proxy for Netbox Docker, problems with HTTP 408 Errors after idle time

### DISCUSSION THREAD

**Discussion by saruter** [netbox-community/netbox-docker#575] [Getting Help]
I'm trying to use Apache as a TLS reverse-proxy in front of netbox-docker but having problems with HTTP 408 Errors and intermittent errors concerning static assets which are not found anymore. 
The whole proxy setup is working fine if you are surfing the netbox application but when being idle for about 2-4 minutes, the next request is answered mostly with a HTTP 408 Error (Request Timeout) from Server "Unit/1.22.0" (seen in Response Headers). After doing a clear CTRL-Refresh, everything works fine again until being idle for some minutes.

There are no problems when accessing netbox via its original docker exposed port on the host itself.

I think somehow the connections between Apache and the netbox nginx are getting out of sync. I tried to tinker around with some timeout settings in the Apache ProxyPass Directive, but without success so far. 

This is the config for the Apache vhost concerning netbox (there are other working vhosts for other tools working fine).
```
<VirtualHost *:443>
  ProxyPreserveHost On
  ServerName dev-netbox.intern.domain.com
  SSLEngine on
  
  RequestHeader set X-Forwarded-Proto "https"
  SSLCertificateFile /etc/ssl/certs/dev-netbox.cer
  SSLCertificateKeyFile /etc/ssl/certs/dev-netbox.key

  ProxyPass / http://127.0.0.1:8000/ timeout=120 keepalive=On (tried with our without these timeout/keepalive settings)
  ProxyPassReverse / http://127.0.0.1:8000/

 </VirtualHost>
```

I'm using Apache as reverse proxy for the non-docker installation method with no issues. Just the netbox-docker installation is driving me nuts.

Does anybody have some hints what could be the reason for this behaviour?

Environment:
Server: CentOS 7
Apache: 2.4.6
netbox-docker: 2.11.9

Regards
Michael

**Comment by RSI-VL** [comment]
Is there a solution to this problem? I have same problem.

**Comment by sdargoeuves** [comment]
I am using a similar approach, and I have experienced the same issue as well.

**Comment by abegosum** [comment]
While not a solution, I have identified a workaround.

The 408 errors seem to be related to some interaction between the Python app and the Apache proxy connections. Apache typically reuses these connections for efficiency's sake, but something about that reuse is causing the error. At least in our environment, disabling reuse of proxy connections has caused the issue to go away:

```
ProxyPass '/' 'http://127.0.0.1:8000/' disablereuse=on
```

**Reply by RSI-VL** [reply]
I solved the problem radically by switching to nginx.
